src/unix/fsevents.c

```diff
@@ -291,13 +289,31 @@ static void uv__fsevents_event_cb(ConstFSEventStreamRef streamRef,
 
       memset(event, 0, sizeof(*event));
       memcpy(event->path, path, len + 1);
-
+      
+#ifndef MAC_OS_X_VERSION_10_7
       if ((eventFlags[i] & kFSEventsModified) != 0 &&
           (eventFlags[i] & kFSEventsRenamed) == 0)
         event->events = UV_CHANGE;
       else
         event->events = UV_RENAME;
-
+#else
+      if (handle->type_flag == kUVFSEventsDirectory) {
+        if ((eventFlags[i] & kFSEventsModified) != 0 &&
```

Comment:
The events that are sent when watching a directory seem to be different than for a file. Making the conditional the same for both files and directories was making all of the tests fail for either one or the other (file vs directory). I documented all of the different scenarios I came across, but am having trouble finding it right now.